source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/imu/imu.py

```diff
@@ -0,0 +1,239 @@
+# Copyright (c) 2022-2024, The Isaac Lab Project Developers.
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+import torch
+from collections.abc import Sequence
+from typing import TYPE_CHECKING
+
+import omni.isaac.core.utils.stage as stage_utils
+import omni.physics.tensors.impl.api as physx
+from pxr import UsdPhysics
+
+import omni.isaac.lab.sim as sim_utils
+import omni.isaac.lab.utils.math as math_utils
+from omni.isaac.lab.markers import VisualizationMarkers
+
+from ..sensor_base import SensorBase
+from .imu_data import ImuData
+
+if TYPE_CHECKING:
+    from .imu_cfg import ImuCfg
+
+
+class Imu(SensorBase):
+    """The Inertia Measurement Unit (IMU) sensor.
+
+    The sensor can be attached to any :class:`RigidObject` in the scene. It provides the linear and angular
```

Comment:
NIT: Would be nice to add some more description on the IMU sensor? Also the issue with using imu on a dummy link? The latter is not recommended from PhysX if I got it correctly?